scout/server/blueprints/variants/controllers.py

```diff
@@ -1551,7 +1552,41 @@ def persistent_filter_actions(
         filter_id = request_form.get("filters")
         filter_obj = store.retrieve_filter(filter_id)
         if filter_obj is not None:
-            form = FiltersFormClass(MultiDict(filter_obj))
+
+            def _coerce_formdata(filter: dict) -> MultiDict:
+                """According to wtforms docs https://wtforms.readthedocs.io/en/3.1.x/forms/#the-form-class,
+                formdata passed to Form init needs to be pre-coerced to types. Hence especially the replacement
+                of locale specific chars in numbers on class process_formdata would not be enough for the
+                better_decimal form fields.
+                """
+                for decimal_value_form_field in [
+                    "cadd_score",
```

Comment:
To have them hardcoded like this is a bit dangerous I think. For instance now that we added other filters like the fusion ones, or any other field that we's add in the future won't be covered if we don't remember to update this list. What about looping through the key/values of the filter and fix them if needed instead?

Reply:
Hm, dunno, this is working and not very strange. The alternative would be to hard code the filter types this should go with since the "pre" construction doesn't allow overriding some method on the specific Class. But sure, we can give it a spin, I'm fine with either!